### ISSUE TITLE
Tailwind/PostCSS — Unable to nest media queries

### ISSUE THREAD

**Issue by xjnny** [batchnz/gatsby-starter-batch-prismic#47]
It'd be nice if we could nest media queries within selectors like this:

```
.classname {
  @screen sm {
    @apply bg-green;
}
  @screen lg {
    @apply bg-pink;
  }
}
```

One of the tailwind creators recommendations for this is:
> https://github.com/postcss/postcss-nested
> If you add that plugin after Tailwind, it'll extract the nested media queries Tailwind generates.

there is also another postcss plugin available:
> https://github.com/jonathantneal/postcss-nesting

**Comment by max8hine** [comment]
We are using **PreCSS**, and **PreCSS** is powered by the following plugins (in this order):

1. postcss-extend-rule
2. postcss-advanced-variables
3. postcss-preset-env
4. postcss-atroot
5. postcss-property-lookup
6. **postcss-nested**

**Comment by hatzipanis** [comment]
@max8hine how was this tested?

The `@screen md` is not working for me:

```css
.border-grid {
  @apply overflow-hidden;

  &__items {
    @apply flex flex-wrap list-reset -mr-1 -mb-1;
  }

  &__item {
    @apply flex flex-col items-center justify-center text-center p-8  h-64;
    border-right: 1px dotted config('colors.grey-light');
    border-bottom: 1px dotted config('colors.grey-light');

    @screen md {
      @apply w-1/4;
    }
  }

  &__title {
    @apply text-lg font-bold mb-2;
  }

  &__subtitle {
    @apply flex text-xs text-grey-dark;
  }
}
```

Could you please look at this again?

**Comment by max8hine** [comment]
@xjnny 	@hatzipanis fixed at commit 9de0cc0

### issue:
The issue caused by **PreCSS** and **TailwindCss** in **PostCss** config execution order.

### details:
**TailwindCss** library itself is avoided by **PreCSS** and **StyleLint** before, to avoid **TaiwindCss** linting check and to avoid **PreCss** intercepts the **TailwindCss**. The trade-off is we could not write in Scss fasion when using **TailwindCss** method.

### Testing:

- [ ] Development Test

- [ ] Build Test (equivalent to production)





``` javascript
postCssPlugins: [
  require('tailwindcss')('./tailwind.config.js'), // current order
  require('precss'),
  require('stylelint'),
  require('autoprefixer')(),
  require('postcss-reporter')({ clearReportedMessages: true }),
]
```

``` javascript
postCssPlugins: [
  require('precss'),
  require('stylelint'),
  require('tailwindcss')('./tailwind.config.js'), // old order
  require('autoprefixer')(),
  require('postcss-reporter')({ clearReportedMessages: true }),
]
```

**Comment by max8hine** [comment]
commit f5fe5ec

moved **Autoprefixer** to the bottom but before **postCssReporter** to supervise all the changes.

Testing:

Adding `appearance: button` into a class to check **Autoprefixer** working result.

- [ ] Development Test
- [ ] Build Test (equivalent to production)

**Comment by hatzipanis** [comment]
Unfortunately this change has broken [config()](https://tailwindcss.com/docs/functions-and-directives/#config) @max8hine. 

Could you please continue to research a solution for nested media queries and ensure you test Tailwind functionality thoroughly.

**Comment by hatzipanis** [comment]
Resolved by #53
